spec/lib/whitehall_importer/create_edition_spec.rb

```diff
@@ -111,13 +117,27 @@
 
         expect(edition.timeline_entries.first).to be_whitehall_migration
         expect(edition.timeline_entries.first.details).to be_first_created
-        expect(edition.timeline_entries.first.created_at).to eq(2.days.ago)
+        expect(edition.timeline_entries.first.created_at).to eq(5.days.ago)
         expect(edition.timeline_entries.first.created_by_id).to eq(first_user.id)
 
-        expect(edition.timeline_entries.last).to be_whitehall_migration
-        expect(edition.timeline_entries.last.details).to be_published
-        expect(edition.timeline_entries.last.created_at).to eq(1.day.ago)
-        expect(edition.timeline_entries.last.created_by_id).to eq(second_user.id)
+        expect(edition.timeline_entries.second).to be_whitehall_migration
+        expect(edition.timeline_entries.second.details).to be_published
+        expect(edition.timeline_entries.second.created_at).to eq(4.days.ago)
+        expect(edition.timeline_entries.second.created_by_id).to eq(second_user.id)
+
+        expect(edition.timeline_entries.third).to be_whitehall_migration
+        expect(edition.timeline_entries.third.details).to be_internal_note
+        expect(edition.timeline_entries.third.created_at).to eq(2.days.ago)
+        expect(edition.timeline_entries.third.created_by_id).to eq(first_user.id)
+
+        expect(edition.timeline_entries.fourth).to be_whitehall_migration
+        expect(edition.timeline_entries.fourth.details).to be_fact_check_request
+        expect(edition.timeline_entries.fourth.created_at).to eq(3.days.ago)
+        expect(edition.timeline_entries.fourth.created_by_id).to eq(second_user.id)
+
+        expect(edition.timeline_entries.fifth).to be_whitehall_migration
+        expect(edition.timeline_entries.fifth.details).to be_fact_check_response
+        expect(edition.timeline_entries.fifth.created_at).to eq(1.day.ago)
```

Comment:
This looks a bit like we're testing lots of cases all in a single test whereas I think we probably want to have context or describe blocks to break these things up, as it seems far more than we can cover with `it "sets the timeline history"`


for example:

`context "when the edition has editorial remarks"` or `describe "editorial remarks"`

It also seems a bit of a smell to have quite as many expectations as we have going on and I wonder if we're missing some assertions (such as the text of internal notes and fact checks).

You might be able to switch some out like:

```
timeline_entry = edition.timeline_entries.third
expect(timeline_entry.attributes).to match a_hash_including(type: :whitehall_migration, created_at: 2.days.ago, created_by_id: first_user.id)
expect(timeline_entry.details.attributes).to match a_hash_including(type: :internal_note, body: a_hash_including({ "body" => "note" })
```

Reply:
Have updated as requested.